### PR TITLE
Allow affine 'shear' tolerance in LocalTracking

### PR DESCRIPTION
**Pull request by arokem**
This supersedes #1043

### CODE REVIEW

File: dipy/tracking/local/tests/test_local_tracking.py [hunk 1]
```diff
@@ -158,6 +162,27 @@ def get_direction(self, p, d):
...
+                                      seeds=seeds,
+                                      affine=funky_affine,
+                                      step_size=1.,
+                                      return_all=True)
+
+    streamlines_funky = iter(streamlines_funky)
+    # We do the same checks here:
+    y = 0
+    sl = next(streamlines_funky)
+    npt.assert_equal(sl[0], [0, y, 0])
```
Comment (gabknight) [inline dipy/tracking/local/tests/test_local_tracking.py]: to get the same results with a different affine, you need to apply the inverse transform to the seed positions, then the inverse affine transformation the the generated streamlines points. The step size must also be adjusted to the voxel size to obtain the same tracking points. 

We miss testing for various affine cases here.. I can jump into it if you want.

Comment (arokem) [inline dipy/tracking/local/tests/test_local_tracking.py]: Hey @gabknight : yes -- if you have a good understanding of what's going on here, your contribution here would be *very* welcome. Thanks!

**Comment by arokem** [comment]
Hey @gabknight : have you had a chance to take a look at this? Someone just raised a similar issue on gitter.

**Comment by gabknight** [comment]
@arokem, not yet. I'm working a PR for the particle filtering tractography. I check this one just after.

**Comment by arokem** [comment]
Hey @gabknight : this was just brought up on the mailing list. Do you have some time to take a look?

**Comment by gabknight** [comment]
@arokem, yes, I will have a look this week and write the tests for various affine matrices. Commit 59405d2f2060c7c000e2b9edd4d711d877150653 could then be add on top of it.

**Comment by arokem** [comment]
Wonderful. Thanks!

**Comment by arokem** [comment]
Closing in favor of #1159

**Pull request closed without merging**